ocrd/ocrd/decorators/__init__.py

```diff
@@ -46,6 +52,40 @@ def ocrd_cli_wrap_processor(
             list_resources=list_resources
         )
         sys.exit()
+    if server_ip or server_port:
+        # IP provided without port
+        if server_ip and not server_port:
+            raise click.UsageError('--server-port is missing.')
+
+        # Port is provided without IP
+        if server_port and not server_ip:
+            raise click.UsageError('--server-ip is missing.')
+
+        # IP and port but without database
+        if server_ip and server_port and not mongo_url:
+            raise click.UsageError('--mongo-url is missing.')
+
+        # Proceed when both IP and port are provided
+        initLogging()
+
+        # Init a processor instance to get access to its information
+        processor = processorClass(workspace=None)
+
+        # Set collection name to the processor name
+        Config.collection_name = processor.ocrd_tool['executable']
+
+        # Set other meta-data
+        Config.processor_class = processorClass
+        Config.title = processor.ocrd_tool['executable']
+        Config.description = processor.ocrd_tool['description']
+        Config.version = processor.version
+        Config.ocrd_tool = processor.ocrd_tool
+        Config.db_url = mongo_url
+
+        # Start the server
+        from ocrd.server.main import app
+        import uvicorn
+        uvicorn.run(app, host=server_ip, port=server_port, access_log=False)
```

Comment:
That's a strange pattern. IIUC you use these class attributes as a kind of cross-module global.

I know it's hard communicating Python objects from your existing runtime context (after parsing the OCR-D CLI) into the web server (I believe Uvicorn offers the same as Unicorn, including inheritance from BaseApplication...). But is that the proper Pythonic way?